src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convert(List<String> lines) {
+        return lines.stream()
+                .map(this::convertToFruitTransaction)
+                .toList();

```

Comment:
It's important to validate the data before converting it. If the input lines contain invalid data, it can lead to runtime exceptions or incorrect behavior. Consider adding validation logic to check if the line has the correct format and the correct number of fields before proceeding with the conversion.